build.rs

```diff
@@ -148,12 +165,21 @@ fn main() {
     println!("cargo:rerun-if-env-changed=LIBPCAP_LIBDIR");
     println!("cargo:rerun-if-env-changed=LIBPCAP_VER");
 
-    let mut libdirpath: Option<PathBuf> = None;
-    if let Ok(libdir) = env::var("LIBPCAP_LIBDIR") {
+    // If user explicitly set LIBPCAP_LIBDIR, honour their wishes. This keeps
+    // existing build scripts running. If it's not set, try pkg-config. If
+    // that's not set, try last ditch effort to build even though library wasn't
+    // explicitly given.
+    let version = if let Ok(libdir) = env::var("LIBPCAP_LIBDIR") {
         println!("cargo:rustc-link-search=native={}", libdir);
-        libdirpath = Some(PathBuf::from(&libdir));
-    }
+        get_libpcap_version(Some(PathBuf::from(&libdir))).unwrap()
+    } else if let Ok(library) = pkg_config::Config::new()
+        .atleast_version(&Version::LOWEST_SUPPORTED.to_string())
+        .probe("libpcap")
+    {
+        Version::parse(&library.version).unwrap()
```

Comment:
Shouldn't this also have something like:

```rust
for link_path in library.link_paths {
  println!("cargo:rustc-link-search=native={}", link_path);
}
```

Reply:
Doesn't look like it:

> After running pkg-config all appropriate Cargo metadata will be printed on stdout if the search was successful.

From https://docs.rs/pkg-config/latest/pkg_config/